### ISSUE TITLE
ARAGORN removing parts of strider's message when empty?

### ISSUE THREAD

**Issue by cbizon** [ranking-agent/aragorn#13]
Query:
```
{
    "message": {
        "query_graph": {
            "edges": {
                "e01": {
                    "object": "n0",
                    "subject": "n1",
                    "predicates": [
                        "biolink:negatively_regulates_entity_to_entity"
                    ]
                }
            },
            "nodes": {
                "n0": {
                    "ids": [
                        "NCBIGene:23221"
                    ],
                    "categories": [
                        "biolink:Gene"
                    ]
                },
                "n1": {
                    "categories": [
                        "biolink:Gene"
                    ]
                }
            }
        }
    }
}
```

Sent directly to strider, this produces no results, but this message:
```
{
    "query_graph": {
        "nodes": {
            "n1": {
                "ids": null,
                "categories": [
                    "biolink:Gene"
                ],
                "is_set": false,
                "constraints": null
            },
            "n0": {
                "ids": [
                    "NCBIGene:23221"
                ],
                "categories": [
                    "biolink:Gene"
                ],
                "is_set": false,
                "constraints": null
            }
        },
        "edges": {
            "e01": {
                "subject": "n1",
                "object": "n0",
                "predicates": [
                    "biolink:negatively_regulates_entity_to_entity"
                ],
                "relation": null,
                "constraints": null
            }
        }
    },
    "knowledge_graph": {
        "nodes": {},
        "edges": {}
    },
    "results": []
}
```

But when we call aragorn, we only get back the qg in the message:
```
{
    "query_graph": {
        "nodes": {
            "n0": {
                "ids": [
                    "NCBIGene:23221"
                ],
                "categories": [
                    "biolink:Gene"
                ],
                "is_set": false
            },
            "n1": {
                "categories": [
                    "biolink:Gene"
                ],
                "is_set": false
            }
        },
        "edges": {
            "e01": {
                "subject": "n1",
                "object": "n0",
                "predicates": [
                    "biolink:negatively_regulates_entity_to_entity"
                ]
            }
        }
    }
}
```

This is legal trapi, but it creates problems for downstream components (like arax) that don't expect this.  Also, I don't see why we would do it.

**Comment by PhillipsOwen** [comment]
do you happen to know what arax would like to see?

is it a totally empty result or just return everything strider sent back?

**Comment by cbizon** [comment]
I'd just pass along everything strider returned.  But especially an empty knowledge graph and results.

**Comment by PhillipsOwen** [comment]
if i throw bad trapi at strider (categories -> categoriesx) get a 200 response and this:

    "knowledge_graph": {
      "nodes": {},
      "edges": {}
    },
    "results": []
  },
  "logs": [],
  "status": null
  
  right now i am sending a 200 and the input json with an additional log item that indicates no results from strider.
  
  should i be returning something other that a 200 to indicate the failure?

**Comment by cbizon** [comment]
If you get back a 200 and an empty result, you don't in general have any way to know that something went wrong.  So you should return the 200.  Returning no results is not a failure, it might be completely correct. 

If strider is returning a 200 on invalid trapi, then that sounds like a bug that should be opened on the strider repo.

**Comment by PhillipsOwen** [comment]
corrected in the the PR #14 merge listed above.
